### PR TITLE
feat: final opcode tweaks

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Some final opcode tweaks:
1. Difficulty always returns ZERO.
2. Coinbase always returns SequencerFeeVault.address.

**Metadata**
- Fixes ENG-1366
- Fixes ENG-1367

### CODE REVIEW

File: l2geth/core/evm.go [hunk 1]
```diff
@@ -65,10 +65,10 @@ func NewEVMContext(msg Message, header *types.Header, chain ChainContext, author
 			Transfer:        Transfer,
 			GetHash:         GetHashFn(header, chain),
 			Origin:          msg.From(),
-			Coinbase:        beneficiary,
+			Coinbase:        common.HexToAddress("0x4200000000000000000000000000000000000011"), // Coinbase is the fee vault.
```
Comment (tynes) [inline l2geth/core/evm.go:68]: This should be defined in a single place so it doesn't need to have the string parsed each time

Comment (smartcontracts) [reply l2geth/core/evm.go:68]: Any other notes? Or should be good for approval once this is fixed?

**Comment by tynes** [comment]
It would be far more ideal to have unit tests in geth for these kinds of things rather than integration tests

File: l2geth/core/evm.go (hunk 1 above)
Comment (tynes) [reply l2geth/core/evm.go:68]: If you think its possible to add some unit tests in geth for this then that would be the most ideal, but I'd approve it if you think its going to be too big of a pain.

**Review by tynes — APPROVED** [review]

**Pull request merged**